cirq/circuits/circuit_test.py

```diff
@@ -2113,3 +2113,45 @@ def test_save_qasm():
 
 x q[0];
 """)
+
+
+def test_submoments():
+    a, b, c, d = cirq.LineQubit.range(4)
+    circuit = cirq.Circuit.from_ops(
+        cirq.H.on(a),
+        cirq.H.on(d),
+        cirq.CZ.on(a, d),
+        cirq.CZ.on(b, c),
+        cirq.CZ.on(a, c),
+        cirq.CZ.on(b, d),
+        cirq.H.on(b),
+        cirq.H.on(c),
+    )
+
+    assert circuit.to_text_diagram() == """
```

Comment:
Reminder to myself: These should use cirq.testing.assert_same_diagram from https://github.com/quantumlib/Cirq/pull/917.